### PR TITLE
Issue #1186 Close file descriptor to make rename work on Windows

### PR DESCRIPTION
**Pull request by praveenkumar**

### CODE REVIEW

File: pkg/minikube/cluster/cluster.go [hunk 1]
```diff
@@ -282,7 +283,19 @@ func (m *MachineConfig) CacheMinikubeISOFromURL() error {
...
-	if err = os.Rename(tmpISOFile.Name(), out.Name()); err != nil {
+	if err = renameFile(tmpISOFile, out); err != nil {
+		return nil
+	}
+
+	return nil
+}
+
+func renameFile(oldFile, newFile *os.File) error {
+	// File descriptor need to be closed otherwise it will throw error
```
Comment (LalatenduMohanty) [inline pkg/minikube/cluster/cluster.go]: I am not sure why we need another function to close the file descriptor. Whether we close it in the same function or in another function it is the same thing.

Comment (praveenkumar) [inline pkg/minikube/cluster/cluster.go]: @LalatenduMohanty This function can be used by other functions if required in future. Putting everything in  a single function should be avoided where it make sense.

Comment (gbraad) [inline pkg/minikube/cluster/cluster.go]: I also think it is on the safer side, as the function forces the closes on function entry. +1

**Comment by gbraad** [comment]
CentOS CI still did not pick this build up... :-s

File: pkg/minikube/cluster/cluster.go (hunk 1 above)
Comment (LalatenduMohanty) [inline pkg/minikube/cluster/cluster.go]: @praveenkumar At this moment the function is a private function, so limited to current package.  Also we are trying not to add more code to minikube package.  I am not sure what should be the right approach here. 
I think you are thinking that we should reuse the function in Minishift as it is easy to do the mistake what we we did last time.

Comment (gbraad) [inline pkg/minikube/cluster/cluster.go]: refactor from private to public is a minor refactor. It should not block this from being integrated...

**Pull request merged**